cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py

```diff
@@ -158,3 +162,53 @@ def validate_subject_ids_in_case(case: TomteCase) -> list[SubjectIdSameAsCaseNam
             error = SubjectIdSameAsCaseNameError(case_name=case.name, sample_name=sample.name)
             errors.append(error)
     return errors
+
+
+def validate_concentration_in_case(case: TomteCase, store: Store):
+    errors = []
+    for sample in case.samples:
+        if has_sample_invalid_concentration(sample=sample, store=store):
+            error = create_invalid_concentration_error(
+                case_name=case.name, sample=sample, store=store
+            )
+            errors.append(error)
+    return errors
+
+
+def create_invalid_concentration_error(case_name: str, sample: TomteSample, store: Store):
+    application: Application = store.get_application_by_tag(sample.application)
+    is_cfdna = is_sample_cfdna(sample)
+    allowed_interval = get_concentration_interval(application=application, is_cfdna=is_cfdna)
+    return InvalidConcentrationIfSkipRCError(
+        case_name=case_name, sample_name=sample.name, allowed_interval=allowed_interval
+    )
+
+
+def has_sample_invalid_concentration(sample: TomteSample, store: Store) -> bool:
+    application: Application | None = store.get_application_by_tag(sample.application)
+    return not is_sample_concentration_allowed(sample=sample, application=application)
+
+
+def is_sample_concentration_allowed(sample: TomteSample, application: Application):
+    concentration = sample.concentration_ng_ul
+    is_cfdna = is_sample_cfdna(sample)
+    interval = get_concentration_interval(application=application, is_cfdna=is_cfdna)
+    return is_sample_concentration_within_interval(concentration=concentration, interval=interval)
+
+
+def is_sample_cfdna(sample: TomteSample):
+    source = sample.source
+    return source == SourceType.CELL_FREE_DNA
```

Comment:
oof, I hate that we have to implement this logic 😢  I wonder if it is even being used. No comment on the implementation though. Just wanted to ventilate

Reply:
It was a quite recent addition with the skip rc flow, so I would guess that there has been use cases for it...